src/mongodb.appup.src

```diff
@@ -1,184 +1,95 @@
 %% -*- mode: erlang -*-
-{"3.0.17",
- [{"3.0.16", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.15", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.14", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.13", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mc_auth_logic, brutal_purge, soft_purge, []},
-    {load_module, mc_worker, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []},
-    {load_module, mc_util, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.12", [
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
-    {load_module, mc_auth_logic, brutal_purge, soft_purge, []},
-    {load_module, mc_worker, brutal_purge, soft_purge, []},
-    {load_module, mc_util, brutal_purge, soft_purge, []},
-    {load_module, mc_topology_logics, brutal_purge, soft_purge, []},
-    {load_module, mc_topology, brutal_purge, soft_purge, []}
-  ]},
-  {"3.0.11", [
+{"3.0.18", [
+   {<<"3.0.[0-7]">>, [
+    {restart_application, mongodb}
+   ]},
+   {<<"3.0.([89]|10)">>, [
     {add_module, mc_util},
-    {load_module, mongoc, brutal_purge, soft_purge, []},
-    {load_module, mc_worker_api, brutal_purge, soft_purge, []},
-    {load_module, mongo_api, brutal_purge, soft_purge, []},
+    {add_module, mc_worker_pid_info},
+    {update, mc_super_sup, supervisor},
```

Comment:
I _think_ (not sure) that this just updates the child specs and supervisor options, but maybe it doesn't spawn missing/new children?  If that's the case, we may need to add something like `{apply, {mc_super_sup, ensure_mc_worker_pid_info, []}}`.

Reply:
Thanks for the comment. I think it should work even if the `mc_worker_pid_info` server is not started (i.e., existing connection should still function and everything should work as before but we can still not use mongodb instances that does not support the old protocol). I think we can leave it as it is for now and then add something like this if testing reveals that it is necessary. Do you think that is ok?

Reply:
I have tried to fix the issue in a new commit below. Please have a look.